packages/lib/action/service.ts

```diff
@@ -136,53 +142,80 @@ export const createAction = async (data: TActionInput): Promise<TAction> => {
   };
 };
 
-export const getActionCountInLastHour = async (actionClassId: string): Promise<number> => {
-  validateInputs([actionClassId, ZId]);
-  try {
-    const numEventsLastHour = await prisma.event.count({
-      where: {
-        eventClassId: actionClassId,
-        createdAt: {
-          gte: new Date(Date.now() - 60 * 60 * 1000),
-        },
-      },
-    });
-    return numEventsLastHour;
-  } catch (error) {
-    throw error;
-  }
-};
+export const getActionCountInLastHour = async (actionClassId: string): Promise<number> =>
+  unstable_cache(
+    async () => {
+      validateInputs([actionClassId, ZId]);
 
-export const getActionCountInLast24Hours = async (actionClassId: string): Promise<number> => {
-  validateInputs([actionClassId, ZId]);
-  try {
-    const numEventsLast24Hours = await prisma.event.count({
-      where: {
-        eventClassId: actionClassId,
-        createdAt: {
-          gte: new Date(Date.now() - 24 * 60 * 60 * 1000),
-        },
-      },
-    });
-    return numEventsLast24Hours;
-  } catch (error) {
-    throw error;
-  }
-};
+      try {
+        const numEventsLastHour = await prisma.event.count({
+          where: {
+            eventClassId: actionClassId,
+            createdAt: {
+              gte: new Date(Date.now() - 60 * 60 * 1000),
+            },
+          },
+        });
+        return numEventsLastHour;
+      } catch (error) {
+        throw error;
+      }
+    },
+    [`getActionCountInLastHour-${actionClassId}`],
+    {
+      tags: [actionClassCache.tag.byActionClassId(actionClassId)],
```

Comment:
As we are getting the actions, shouldn't that be a cache `actionCache.tag.byActionClassId(actionClassId)`?

Reply:
but when is this invalidated? it will probably be invalidated in every place actionClassCache is invalidated, which is a duplicate.